src/allmydata/util/cputhreadpool.py

```diff
@@ -46,14 +46,37 @@
 P = ParamSpec("P")
 R = TypeVar("R")
 
+# Is running in a thread pool disabled? Should only be true in synchronous unit
+# tests.
+_DISABLED = False
+
 
 def defer_to_thread(
-    reactor: IReactorFromThreads, f: Callable[P, R], *args: P.args, **kwargs: P.kwargs
+    f: Callable[P, R], *args: P.args, **kwargs: P.kwargs
 ) -> Deferred[R]:
     """Run the function in a thread, return the result as a ``Deferred``."""
+    if _DISABLED:
+        return maybeDeferred(f, *args, **kwargs)
+
     # deferToThreadPool has no type annotations...
     result = deferToThreadPool(reactor, _CPU_THREAD_POOL, f, *args, **kwargs)
     return cast(Deferred[R], result)
 
 
-__all__ = ["defer_to_thread"]
+def disable_thread_pool_for_test(test: TestCase) -> None:
```

Comment:
The comment above `_DISABLED` would probably be a useful addition to this function's docstring.  Otherwise, it's not immediately obvious why this would be desired.

Even so, what if the code doesn't work when work is dispatched to the thread pool?  Wouldn't it be better to exercise that in the test suite and catch problems?  Should we instead convert the synchronous tests to asynchronous tests?  This might also have the advantage of avoiding the need for this global mutable state.

If none of that makes sense, then maybe it at least makes sense to make this into test infrastructure and have it happen automatically whenever a `SyncTestCase` is running, instead of having each test case subclass manually call this.